### ISSUE TITLE
Bug: temp-muted role overwritten in PRC when connected to a PR

### ISSUE THREAD

**Issue by DarkWolf009** [pinano-discord/Pinano-Discord-Bot#44]
Repro: Assign "temp-muted" role to self (this will only work correctly if you are able to assign roles (can ask someone to do it for you) and you are not a mod (mods are not affected by the role iirc)) - join voice channel - you will be able to type in the practice room chat despite being temp muted.

This is because of the bot's permission granting method, which acts as an override. To fix this, the bot must check for the "temp-muted" role on "guildMemberUpdate" and apply the override based on the result.

It could look something like this (dummy code provided by Kai):
```
client.on('guildMemberUpdate', (oldMember, newMember) =>{
    if(newMember.roles.map(res => res.name).includes('temp muted')) {
        //user now has muted role
        prChan.overwritePermissions(newMember.user, { SEND_MESSAGES: false }); 
    }
    if(oldMember.roles.map(res => res.name).includes('temp muted')){
        //user now has their muted role removed
        prChan.overwritePermissions(oldMember.user, { SEND_MESSAGES: true }); 
    }
});```

**Comment by pianiststickman** [comment]
n.b. we would probably need to fix up the `voiceStatusUpdate` event handler as well and add a role check as another condition on allowing PRC privileges. In other words, if they're in PRC already and they get the role added, we need to remove it - that's what the above does. If they have the role added and they land themselves in PRC, we need to not add it like we normally would.

**Comment by pianiststickman** [comment]
Fixed in #77
